### ISSUE TITLE
really slow trying to bring up forge in large repo

### ISSUE THREAD

**Issue by nicklan** [magit/forge#232]
With a large repo, after finally finishing `forge-pull` by trying over and over to get past rate limits, hitting `f` now makes emacs hang for about 10 minutes while the forge info inserts into the buffer.  I suspect this is due to the repo having a TON of tags (26766 to be exact).  I've cut my status buffer way down to:
```elisp
(setq magit-status-sections-hook
      '(magit-insert-untracked-files
        magit-insert-unstaged-changes
        magit-insert-staged-changes
        magit-insert-stashes))
```

but can't figure out a way to tell forge not to include branches and tags.  Here's what the buffer looks like after I hit `f` and wait:
![magit1](https://user-images.githubusercontent.com/178912/70745341-6cacd080-1cd8-11ea-9dcc-a589e80e6087.png)

I mostly just want to use forge to create pull requests, is there a way I can disable all the extra info?  Thanks!

**Comment by tarsius** [comment]
Instrument both `magit-` and `forge-` using `elp-instrument-package`, invoke `magit-status` and then post the result which you can get with `elp-results`.

> can't figure out a way to tell forge not to include branches and tags

Where does forge "include branches and tags"? I suspect you would get the same issue with magit without forge. (Which is why you should instrument both packages.)

**Comment by nicklan** [comment]
thanks.  I think you're right that it's not actually forge, but rather magit.  let me know if you want me to move this issue over there.  here are the results:
```
magit-setup-buffer-internal                       2           88.590047829  44.295023914
magit-refresh-buffer                              2           88.391402344  44.195701172
magit-show-refs                                   1           87.974854368  87.974854368
magit-refs-setup-buffer                           1           87.954813559  87.954813559
magit-refs-refresh-buffer                         1           87.814745235  87.814745235
magit-run-section-hook                            2           70.60084516   35.30042258
magit-insert-tags                                 1           59.737846494  59.737846494
magit-insert-heading                              28943       38.921305220  0.0013447571
magit-section-show                                6           28.950912226  4.8251520376
magit-section-hide                                28937       14.403978815  0.0004977702
magit-section-maybe-update-visibility-indicator   28943       10.727737923  0.0003706505
magit-insert-remote-branches                      1           10.240417465  10.240417465
magit-get-section                                 28951       9.1730142040  0.0003168461
magit-section-ident                               173686      3.5803221330  2.061...e-05
magit-section-match-1                             345186      3.4281567249  9.931...e-06
magit-maybe-make-margin-overlay                   28944       2.0785905609  7.181...e-05
magit-section-match                               57889       2.0241238140  3.496...e-05
magit-section-cached-visibility                   28951       0.9075601189  3.134...e-05
magit-process-file                                39          0.8816433539  0.0226062398
magit-section-match-2                             287303      0.8380082069  2.916...e-06
magit-git-insert                                  22          0.799347876   0.0363339943
magit-section-ident-value                         1166373     0.7564672439  6.485...e-07
magit-status                                      1           0.635277902   0.635277902
magit-status-setup-buffer                         1           0.635271657   0.635271657
magit-status-refresh-buffer                       1           0.551721395   0.551721395
magit-section                                     28951       0.4641428480  1.603...e-05
forge-get-repository                              10          0.447141079   0.0447141079
magit-git-items                                   5           0.429837188   0.0859674376
magit-insert-untracked-files                      1           0.419073317   0.419073317
magit-git-lines                                   8           0.34443593    0.0430544912
magit-refs--format-focus-column                   28936       0.2731970720  9.441...e-06
magit-section-match-assoc                         26825       0.2150786469  8.017...e-06
magit-refs--insert-cherry-commits                 28936       0.1994468689  6.892...e-06
magit--overlay-at                                 28941       0.1434936920  4.958...e-06
magit-refs--insert-refname-p                      28936       0.1261108789  4.358...e-06
forge-bug-reference-setup                         2           0.1111395229  0.0555697614
magit-insert-local-branches                       1           0.092143822   0.092143822
magit-insert-child-count                          28949       0.0890951389  3.077...e-06
magit-refs-mode                                   1           0.08840276    0.08840276
magit-status-mode                                 1           0.080686229   0.080686229
magit-diff-expansion-threshold                    28951       0.0650630469  2.247...e-06
forge-insert-pullreqs                             1           0.059818756   0.059818756
magit-rev-parse-safe                              21          0.0575579949  0.0027408569
magit-git-str                                     21          0.057473489   0.0027368328
magit-section-maybe-cache-visibility              28943       0.0541952889  1.872...e-06
magit-toplevel                                    16          0.0537851910  0.0033615744
magit-log-propertize-keywords                     28942       0.0427293050  1.476...e-06
magit-get-mode-buffer                             3           0.040539711   0.0135132369
magit-file-section-p                              28951       0.0336551970  1.162...e-06
magit-mode                                        2           0.029286645   0.0146433225
magit-insert-unstaged-changes                     1           0.026864538   0.026864538
magit-git-exit-code                               1           0.025530972   0.025530972
magit-show-refs-arguments                         1           0.020031941   0.020031941
magit-refs--propertize-branch                     2179        0.0161920550  7.430...e-06
magit-get-all                                     13          0.015647403   0.0012036463
magit-git-string                                  8           0.013688626   0.0017110782
magit-get                                         11          0.013198839   0.0011998944
magit-insert-staged-changes                       1           0.012628523   0.012628523
magit-config-get-from-cached-list                 11          0.0108406929  0.0009855175
magit-list-remotes                                4           0.010824945   0.0027062362
magit-refs--format-local-branches                 1           0.010473466   0.010473466
forge--get-remote                                 3           0.0095539099  0.0031846366
magit-maybe-save-repository-buffers               2           0.007438631   0.0037193155
magit-save-repository-buffers                     2           0.007408687   0.0037043435
magit-insert-stashes                              1           0.007375051   0.007375051
magit-git-wash                                    3           0.005728781   0.0019095936
magit-insert-branch-description                   1           0.004609596   0.004609596
magit-rev-parse                                   3           0.004358801   0.0014529336
magit-log-wash-log                                1           0.004059925   0.004059925
magit--propertize-face                            2190        0.0029814210  1.361...e-06
forge-sql                                         3           0.002827161   0.0009423869
magit-abbrev-length                               1           0.002808068   0.002808068
magit-rev-verify                                  2           0.002671291   0.0013356455
magit-git-string-p                                2           0.002667063   0.0013335315
magit-section-update-highlight                    7           0.002484827   0.0003549752
magit-load-config-extensions                      2           0.002480437   0.0012402185
magit-section-highlight                           6           0.002104009   0.0003506681
magit-section-make-overlay                        6           0.002041632   0.0003402719
magit--put-face                                   2617        0.0018673080  7.135...e-07
magit-bare-repo-p                                 1           0.001734386   0.001734386
magit-libgit-repo                                 1           0.001702532   0.001702532
magit-display-buffer                              2           0.001461847   0.0007309235
magit-display-buffer-traditional                  2           0.001388662   0.000694331
magit-refs--format-local-branch                   60          0.001356024   2.260...e-05
magit-git-dir                                     3           0.0013099880  0.0004366626
magit-wash-sequence                               1           0.001237975   0.001237975
magit-gitdir                                      1           0.001235565   0.001235565
magit-log-wash-rev                                6           0.00122936    0.0002048933
magit-get-current-branch                          2           0.0010883339  0.0005441669
magit-preserve-section-visibility-cache           1754        0.0010014310  5.709...e-07
magit-buffer-margin-p                             2125        0.0006365970  2.995...e-07
magit--safe-default-directory                     9           0.000541696   6.018...e-05
magit-unhighlight-region                          52          0.000496039   9.539...e-06
forge--split-url                                  3           0.000490384   0.0001634613
magit-process-environment                         39          0.0004105139  1.052...e-05
magit-file-accessible-directory-p                 9           0.0003624730  4.027...e-05
forge--url-regexp                                 3           0.0003607070  0.0001202356
magit-diff-highlight                              6           0.000245963   4.099...e-05
magit-process-git-arguments                       39          0.0001973630  5.060...e-06
magit--process-coding-system                      39          0.0001899420  4.870...e-06
forge-db                                          6           0.00014172    2.361...e-05
magit-diff-scope                                  6           0.0001363149  2.271...e-05
magit-section-goto-successor                      2           0.00013224    6.612e-05
magit-rebase-in-progress-p                        1           9.1717e-05    9.1717e-05
magit-make-margin-overlay                         26          8.985e-05     3.455...e-06
magit-delete-region-overlays                      52          8.8237e-05    1.696...e-06
magit-log-format-margin                           6           8.2027e-05    1.367...e-05
magit-log-format-author-margin                    6           6.6912e-05    1.1152e-05
magit-expand-git-file-name                        11          6.689...e-05  6.081...e-06
magit-pre-command-hook                            42          6.056e-05     1.441...e-06
magit-repository-local-get                        4           5.0138e-05    1.25345e-05
magit-diff-type                                   6           4.84e-05      8.066...e-06
magit-delete-line                                 6           4.760...e-05  7.934...e-06
magit-set-buffer-margin                           2           3.9958e-05    1.9979e-05
magit-repository-local-exists-p                   4           3.6824e-05    9.206e-06
forge-insert-issues                               1           3.457e-05     3.457e-05
magit-refresh-get-relative-position               2           2.9876e-05    1.4938e-05
magit-current-section                             15          2.976...e-05  1.9844e-06
forge--split-url-path                             3           2.5381e-05    8.460...e-06
magit-file-mode-turn-on                           26          2.4655e-05    9.482...e-07
magit-process-unset-mode-line-error-status        2           2.3978e-05    1.1989e-05
magit-ignore-submodules-p                         1           1.9612e-05    1.9612e-05
magit-save-window-configuration                   2           1.875...e-05  9.377...e-06
magit--age                                        6           1.621...e-05  2.702...e-06
magit-region-sections                             6           1.591e-05     2.651...e-06
magit-section--maybe-wash                         6           1.4538e-05    2.423e-06
magit--assert-default-directory                   1           1.2815e-05    1.2815e-05
magit-gitimpl                                     1           1.1184e-05    1.1184e-05
magit-set-header-line-format                      1           9.637e-06     9.637e-06
magit-set-window-margin                           2           8.318e-06     4.159e-06
magit--tramp-asserts                              1           7.935e-06     7.935e-06
magit-log-margin-width                            2           6.568...e-06  3.284...e-06
magit-repository-local-repository                 4           5.250...e-06  1.312...e-06
magit-diff-use-hunk-region-p                      6           4.687e-06     7.811...e-07
magit-section-maybe-paint-visibility-ellipses     7           4.509e-06     6.441...e-07
magit-margin-option                               8           4.486...e-06  5.607...e-07
magit-maybe-set-dedicated                         2           3.472e-06     1.736e-06
magit-hunk-section-p                              2           3.323e-06     1.6615e-06
magit-diff--get-value                             1           3.137e-06     3.137e-06
magit-xref-setup                                  2           2.287...e-06  1.143...e-06
magit-log--get-value                              1           1.897e-06     1.897e-06
magit--add-face-text-property                     1           1.783e-06     1.783e-06
magit-insert-error-header                         1           7e-07         7e-07
magit--libgit-available-p                         1           3.27e-07      3.27e-07
magit-cancel-section                              2           0             0.0
magit--insert-diff                                2           0             0.0
```

**Comment by tarsius** [comment]
See https://github.com/magit/magit/issues/4017.
